de.dlr.sc.virsat.model.edit/src/de/dlr/sc/virsat/model/concept/migrator/AMigrator.java

```diff
@@ -608,6 +656,13 @@ public EObject getEObject(URI uri, boolean loadOnDemand) {
 		String nsURI = ReleaseUtils.getNamespaceURI(conceptResourceUri);
 		Migrator migrator = MigratorRegistry.getInstance().getMigrator(nsURI);
 		
+		if (migrator == null) {
+			DVLMEditPlugin.getPlugin().getLog().log(new Status(Status.ERROR, DVLMEditPlugin.PLUGIN_ID, 
+					"Could not get DVLM migrator for concept resource: " + conceptResourceUri));
+			//Check that all dependent concepts and their plugins are available in platform...  
+			return new ResourceSetImpl();

```

Comment:
Could you please explain this fragment? When can the migrator be null, why are we returning a new resource set and what does it mean to the calling code?

Reply:
Well, I had the problem that a test case was failing in the maven build (only in the maven build!) because there the migrator was "null". I was debugging that for several hours until I tried returning a simple resource set in that case... The resource set then informed me that a required concepts of that cocept could not be found, because its plugin was not found... So I simply had to add the dependent concept's plugin as required plugin in the test fragement...
If you don't like that we can also remove this fragement... But I was that pissed of that the migrator was just "null" without any explonation that I added this pease of code. I had that pronblem some time ago and I think there that would have helped as well